<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>1. Accessing Human Cell Atlas Data • ExploringHCACxG</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="1. Accessing Human Cell Atlas Data">
<meta property="og:description" content="ExploringHCACxG">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ExploringHCACxG</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A_HCA.html">1. Accessing Human Cell Atlas Data</a>
    </li>
    <li>
      <a href="../articles/B_CXG.html">2. Accessing CellxGene Data</a>
    </li>
    <li>
      <a href="../articles/C_AnVIL.html">3. Accessing Human Cell Atlas Data on the AnVIL Cloud</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mtmorgan/ExploringHCACxG">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>1. Accessing Human Cell Atlas Data</h1>
                        <h4 data-toc-skip class="author">Martin Morgan<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, Maya
McDaniel, Yubo Cheng<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/ExploringHCACxG/blob/HEAD/vignettes/A_HCA.Rmd"><code>vignettes/A_HCA.Rmd</code></a></small>
      <div class="hidden name"><code>A_HCA.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>FIXME: Along with the topic of your workshop, include how students
can expect to spend their time. For the description may also include
information about what type of workshop it is (e.g. instructor-led live
demo, lab, lecture + lab, etc.). Instructors are strongly recommended to
provide completely worked examples for lab sessions, and a set of
stand-alone notes that can be read and understood outside of the
workshop.</p>
</div>
<div class="section level3">
<h3 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h3>
<p>The focus is on the <a href="https://bioconductor.org/packages/hca" class="external-link">hca</a> package. This
package emphasize <a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a> and ‘tidy’
approaches to working with data.frames. Files downloaded from the HCA or
CellXGene web sites can be imported into R / Bioconductor as <a href="https://bioconductor.org/packages/SingleCellExperiment" class="external-link">SingleCellExperiment</a>
objects through the <a href="https://bioconductor.org/packages/LoomExperiment" class="external-link">LoomExperiment</a>
and <a href="https://bioconductor.org/packages/zellkonverter" class="external-link">zellkonverter</a>
packages.</p>
</div>
<div class="section level3">
<h3 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction to the HCA data portal</td>
<td>5m</td>
</tr>
<tr class="even">
<td>HCA discovery in R / Bioconductor</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="workshop-goals-and-objectives">Workshop goals and objectives<a class="anchor" aria-label="anchor" href="#workshop-goals-and-objectives"></a>
</h3>
<p>List “big picture” student-centered workshop goals and learning
objectives. Learning goals and objectives are related, but not the same
thing. These goals and objectives will help some people to decide
whether to attend the conference for training purposes, so please make
these as precise and accurate as possible.</p>
<p><em>Learning goals</em> are high-level descriptions of what
participants will learn and be able to do after the workshop is over.
<em>Learning objectives</em>, on the other hand, describe in very
specific and measurable terms specific skills or knowledge attained. The
<a href="#bloom">Bloom’s Taxonomy</a> may be a useful framework for
defining and describing your goals and objectives, although there are
others.</p>
</div>
<div class="section level3">
<h3 id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h3>
<p>Some examples:</p>
<ul>
<li>describe how to…</li>
<li>identify methods for…</li>
<li>understand the difference between…</li>
</ul>
</div>
<div class="section level3">
<h3 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h3>
<ul>
<li>analyze xyz data to produce…</li>
<li>create xyz plots</li>
<li>evaluate xyz data for artifacts</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="workshop">Workshop<a class="anchor" aria-label="anchor" href="#workshop"></a>
</h2>
<div class="section level3">
<h3 id="single-cell-analysis-in-bioconductor">Single-cell analysis in <em>Bioconductor</em><a class="anchor" aria-label="anchor" href="#single-cell-analysis-in-bioconductor"></a>
</h3>
<p>Extensive support for single-cell analysis in
<em>Bioconductor</em></p>
<ul>
<li>
<p>E.g., 213 packages tagged ‘SingleCell’ in the release branch</p>
<p><img src="images/BiocSingleCell.png"></p>
</li>
<li><p>Quality control, normalization, variance modeling, data
integration / batch correction, dimensionality reduction, clustering,
differential expression, cell type classification, etc</p></li>
<li><p>Infrastructure to support large out-of-memory data, etc.</p></li>
<li><p>Typically, processing after a ‘count matrix’ has been
obtained</p></li>
</ul>
<p>Orchestrating Single-Cell Analysis with <em>Bioconductor</em> (<a href="https://bioconductor.org/books/OSCA/" class="external-link">OSCA</a>) – an amazing
resource!</p>
<p><img src="images/OSCA.png"></p>
</div>
<div class="section level3">
<h3 id="a-quick-dplyr-tidy-refresher">A quick <a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a>
‘tidy’ refresher<a class="anchor" aria-label="anchor" href="#a-quick-dplyr-tidy-refresher"></a>
</h3>
<p>Recall</p>
<ul>
<li>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble()</a></code> is a <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame()</a></code> but with a
nicer display.</li>
<li>A few common commands allow many standard manipulations
<ul>
<li>
<code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter()</a></code>: filter to include only specific rows</li>
<li>
<code><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select()</a></code>: select columns</li>
<li>
<code><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate()</a></code>: change or add columns</li>
<li>
<code>glimpes()</code>: get a quick summary of the first rows of the
<code>tibble</code>
</li>
<li>
<code>left_join(x, y, by =)</code> is a somehat more advanced
function, merging all rows of tibble <code>x</code> with rows in
<code>y</code> that match for columns specified by
<code>by=</code>.</li>
</ul>
</li>
<li>
<em>R</em> introduced the ‘pipe’ <code>|&gt;</code>, typically used
to take the output of one command and ‘pipe’ it to the first input
argument of the next command</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">mtcars</span>, rownames <span class="op">=</span> <span class="st">"car"</span><span class="op">)</span></span>
<span><span class="va">tbl</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 32 × 12</span></span></span>
<span><span class="co">#&gt;    car           mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Mazda RX4    21       6  160    110  3.9   2.62  16.5     0     1     4     4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Mazda RX4 …  21       6  160    110  3.9   2.88  17.0     0     1     4     4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Datsun 710   22.8     4  108     93  3.85  2.32  18.6     1     1     4     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Hornet 4 D…  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Hornet Spo…  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Valiant      18.1     6  225    105  2.76  3.46  20.2     1     0     3     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Duster 360   14.3     8  360    245  3.21  3.57  15.8     0     0     3     4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Merc 240D    24.4     4  147.    62  3.69  3.19  20       1     0     4     2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Merc 230     22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Merc 280     19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 22 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span>
<span><span class="va">tbl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">cyl</span> <span class="op">==</span> <span class="fl">6</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">car</span>, <span class="va">mpg</span>, <span class="va">disp</span>, <span class="va">hp</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>`L/100km` <span class="op">=</span> <span class="fl">235.2</span> <span class="op">/</span> <span class="va">mpg</span><span class="op">)</span> <span class="co"># Liters per 100 km</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 5</span></span></span>
<span><span class="co">#&gt;   car              mpg  disp    hp `L/100km`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Mazda RX4       21    160    110      11.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Mazda RX4 Wag   21    160    110      11.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Hornet 4 Drive  21.4  258    110      11.0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Valiant         18.1  225    105      13.0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Merc 280        19.2  168.   123      12.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Merc 280C       17.8  168.   123      13.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> Ferrari Dino    19.7  145    175      11.9</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="exploring-the-human-cell-atlas-data-portal">Exploring the Human Cell Atlas Data Portal<a class="anchor" aria-label="anchor" href="#exploring-the-human-cell-atlas-data-portal"></a>
</h3>
<p>Human Cell Atlas <a href="https://data.humancellatlas.org/explore/projects" class="external-link">data
portal</a></p>
<p><img src="images/HCADataPortal.png"></p>
</div>
<div class="section level3">
<h3 id="working-with-hca-data">Working with HCA data<a class="anchor" aria-label="anchor" href="#working-with-hca-data"></a>
</h3>
<p>Why use an <em>R</em> script when the Data Portal exists?</p>
<ul>
<li>Easily reproducible</li>
<li>Flexible exploration of rich &amp; complex data</li>
<li>Direct integration with <em>Bioconductor</em> single-cell
workflows</li>
</ul>
<p>Load required packages</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">hca</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LoomExperiment</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Discover projects</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## basic information, like HCA data portal</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 261 × 14</span></span></span>
<span><span class="co">#&gt;    projectId     proje…¹ genus…² sampl…³ speci…⁴ speci…⁵ selec…⁶ libra…⁷ nucle…⁸</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 74b6d569-3b1… 1.3 Mi… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 53c53cd4-812… A Cell… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 7027adc6-c9c… A Cell… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 94e4ee09-9b4… A Huma… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 60ea42e1-af4… A Prot… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ef1e3497-515… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> f86f1ab4-1fb… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 602628d7-c03… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> c1810dbc-16d… A cell… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> c0518445-3b3… A cell… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 251 more rows, 5 more variables: pairedEnd &lt;list&gt;, workflow &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   specimenDisease &lt;list&gt;, donorDisease &lt;list&gt;, developmentStage &lt;list&gt;, and</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   abbreviated variable names ¹​projectTitle, ²​genusSpecies, ³​sampleEntityType,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁴​specimenOrgan, ⁵​specimenOrganPart, ⁶​selectedCellType,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁷​libraryConstructionApproach, ⁸​nucleicAcidSource</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>Columns with type <code>&lt;list&gt;</code> usually have some
projects with 0, 1, 2, or more of the corresponding column. This is more
clearly displayed when the dplyr print method allocates more room per
column, or when using <code><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse()</a></code>, or by pulling the column
from the table.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">specimenOrganPart</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 261 × 1</span></span></span>
<span><span class="co">#&gt;    specimenOrganPart</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> <span style="color: #949494;">&lt;chr [2]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> <span style="color: #949494;">&lt;chr [0]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> <span style="color: #949494;">&lt;chr [0]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> <span style="color: #949494;">&lt;chr [4]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> <span style="color: #949494;">&lt;chr [0]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> <span style="color: #949494;">&lt;chr [3]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> <span style="color: #949494;">&lt;chr [0]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> <span style="color: #949494;">&lt;chr [6]&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 251 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span>
<span><span class="va">p</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 261</span></span>
<span><span class="co">#&gt; Columns: 14</span></span>
<span><span class="co">#&gt; $ projectId                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "74b6d569-3b11-42ef-b6b1-a0454522b4a0", "5…</span></span>
<span><span class="co">#&gt; $ projectTitle                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "1.3 Million Brain Cells from E18 Mice", "…</span></span>
<span><span class="co">#&gt; $ genusSpecies                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "Mus musculus", "Homo sapiens", "Mus musc…</span></span>
<span><span class="co">#&gt; $ sampleEntityType            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "specimens", "specimens", "specimens", &lt;"…</span></span>
<span><span class="co">#&gt; $ specimenOrgan               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "brain", "prostate gland", "heart", "live…</span></span>
<span><span class="co">#&gt; $ specimenOrganPart           <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "cortex", &lt;"peripheral zone of prostate",…</span></span>
<span><span class="co">#&gt; $ selectedCellType            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "neuron", &lt;"basal cell of prostate epithe…</span></span>
<span><span class="co">#&gt; $ libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "10x 3' v2", "10X v2 sequencing", "10X 3'…</span></span>
<span><span class="co">#&gt; $ nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "single cell", "single cell", "single cel…</span></span>
<span><span class="co">#&gt; $ pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, …</span></span>
<span><span class="co">#&gt; $ workflow                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;&gt;, &lt;"optimus_post_processing_v1.0.0", "o…</span></span>
<span><span class="co">#&gt; $ specimenDisease             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "normal", "normal", "normal", &lt;"hepatocel…</span></span>
<span><span class="co">#&gt; $ donorDisease                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "normal", "normal", "normal", &lt;"cholangio…</span></span>
<span><span class="co">#&gt; $ developmentStage            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "mouse embryo stage", "human adult stage"…</span></span>
<span><span class="va">p</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">specimenOrganPart</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] "cortex"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; [1] "peripheral zone of prostate" "transition zone of prostate"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; character(0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span>
<span><span class="co">#&gt; character(0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span>
<span><span class="co">#&gt; [1] "saliva"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[6]]</span></span>
<span><span class="co">#&gt; [1] "epithelium of bronchus"       "epithelium of trachea"       </span></span>
<span><span class="co">#&gt; [3] "inferior nasal concha"        "terminal bronchus epithelium"</span></span></code></pre></div>
<p>The table can be expanded so that there is one row for each
<code>specimenOrganPart</code>, rather than for each project; it is
usually not productive to unnest multiple columns, because this leads to
a combinatorial number of rows.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html" class="external-link">unnest</a></span><span class="op">(</span><span class="va">specimenOrganPart</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 415 × 14</span></span></span>
<span><span class="co">#&gt;    projectId     proje…¹ genus…² sampl…³ speci…⁴ speci…⁵ selec…⁶ libra…⁷ nucle…⁸</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 74b6d569-3b1… 1.3 Mi… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   cortex  &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 53c53cd4-812… A Cell… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   periph… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 53c53cd4-812… A Cell… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   transi… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 60ea42e1-af4… A Prot… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   saliva  &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ef1e3497-515… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   epithe… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ef1e3497-515… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   epithe… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> ef1e3497-515… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   inferi… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> ef1e3497-515… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   termin… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 602628d7-c03… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   annulu… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 602628d7-c03… A Sing… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   nucleu… &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 405 more rows, 5 more variables: pairedEnd &lt;list&gt;, workflow &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   specimenDisease &lt;list&gt;, donorDisease &lt;list&gt;, developmentStage &lt;list&gt;, and</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   abbreviated variable names ¹​projectTitle, ²​genusSpecies, ³​sampleEntityType,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁴​specimenOrgan, ⁵​specimenOrganPart, ⁶​selectedCellType,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁷​libraryConstructionApproach, ⁸​nucleicAcidSource</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>Arguments to <code><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects()</a></code> allow information to be returned
in different formats, or to return much more information about each
project. Using <code>as = "list"</code> produces a nested data structure
with all information provided by the HCA Data Portal; exploring this
data is facilitated by <code>listveiwer::jsonedit()</code>, using
<code>as = "lol"</code> and <code>lol_*</code> functions in the <a href="https://bioconductor.org/packages/hca" class="external-link">hca</a> package, or perhaps
<code><a href="https://rdrr.io/pkg/cellxgenedp/man/jmespath.html" class="external-link">cellxgenedp::jmespath()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## all information, as a list-of-lists</span></span>
<span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects</a></span><span class="op">(</span>as <span class="op">=</span> <span class="st">"list"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">pl</span><span class="op">)</span></span>
<span><span class="co">#&gt; pagination termFacets       hits </span></span>
<span><span class="co">#&gt;          8         36        261</span></span></code></pre></div>
<p>An alternative to working with all projects is to apply filters
during the original query. For instance, to select subsets of projects,
e.g., studying liver. It is possible to specify several filters at once,
as we will see below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">liver_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/filters.html" class="external-link">filters</a></span><span class="op">(</span></span>
<span>    specimenOrgan <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"liver"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">liver_projects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects</a></span><span class="op">(</span><span class="va">liver_filter</span><span class="op">)</span></span>
<span><span class="va">liver_projects</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">projectId</span>, <span class="va">projectTitle</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 21 × 2</span></span></span>
<span><span class="co">#&gt;    projectId                            projectTitle                            </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 94e4ee09-9b4b-410a-84dc-a751ad36d0df A Human Liver Cell Atlas reveals Hetero…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> a9301beb-e9fa-42fe-b75c-84e8a460c733 A human cell atlas of fetal gene expres…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> c31fa434-c9ed-4263-a9b6-d9ffb9d44005 A single-cell atlas of chromatin access…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 8ab8726d-81b9-4bd2-acc2-4d50bee786b4 An organoid and multi-organ development…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 04ad400c-58cb-40a5-bc2b-2279e13a910b Blood and immune development in human f…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> c7c54245-548b-4d4f-b15e-0d7e238ae6c8 Construction of a single-cell transcrip…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> a9f5323a-ce71-471c-9caf-04cc118fd1d7 Decoding Human Megakaryocyte Development</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> f2fe82f0-4454-4d84-b416-a885f3121e59 Decoding human fetal liver haematopoies…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> ccef38d7-aa92-4010-9621-c4c7b1182647 Dissecting the clonal nature of allelic…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 4d6f6c96-2a83-43d8-8fe1-0f53bffd4674 Dissecting the human liver cellular lan…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 11 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span>
<span></span>
<span><span class="co">## facet options available for filtering: facet_options()</span></span></code></pre></div>
<p>Main functions for exploring HCA data</p>
<ul>
<li>
<code><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects()</a></code>: finding projects of interest.</li>
<li>
<code><a href="https://rdrr.io/pkg/hca/man/bundles.html" class="external-link">bundles()</a></code>, <code><a href="https://rdrr.io/pkg/hca/man/samples.html" class="external-link">samples()</a></code>: can be useful when
exploring a single project in detail.</li>
<li>
<code><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files()</a></code>: use to download files (e.g., fastq files to
start an analysis from scratch; ‘.loom’ files containing gene x cell
count matrices produced by standard HCA pipelines).</li>
<li>
<code><a href="https://rdrr.io/pkg/hca/man/hca.html" class="external-link">hca_next()</a></code>, <code><a href="https://rdrr.io/pkg/hca/man/hca.html" class="external-link">hca_prev()</a></code> to ‘page’ through
large datasets.</li>
</ul>
</div>
<div class="section level3">
<h3 id="discovering-and-downloading--loom-files">Discovering and downloading ‘.loom’ files<a class="anchor" aria-label="anchor" href="#discovering-and-downloading--loom-files"></a>
</h3>
<p>HCA standard processing pipelines</p>
<p>Projects with .loom files produced by the standard pipeline satisfy
the following filter</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">standard_loom_file_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/filters.html" class="external-link">filters</a></span><span class="op">(</span></span>
<span>    fileSource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"DCP/2 Analysis"</span><span class="op">)</span>,</span>
<span>    fileFormat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"loom"</span><span class="op">)</span>,</span>
<span>    workflow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"optimus_post_processing_v1.0.0"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">loom_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files</a></span><span class="op">(</span><span class="va">standard_loom_file_filter</span><span class="op">)</span></span></code></pre></div>
<p>Which projects studying the liver have a standard loom file
available?</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## inner_join -- projectId &amp; projectTitles in both tibbles</span></span>
<span><span class="va">liver_loom_projects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span></span>
<span>    <span class="va">liver_projects</span>, <span class="va">loom_files</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"projectId"</span>, <span class="st">"projectTitle"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">liver_loom_projects</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">projectId</span>, <span class="va">fileId</span>, <span class="va">projectTitle</span>, fileSize <span class="op">=</span> <span class="va">size</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11 × 4</span></span></span>
<span><span class="co">#&gt;    projectId                            fileId                   proje…¹ fileS…²</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> f2fe82f0-4454-4d84-b416-a885f3121e59 815e27f4-56f9-538a-a770… Decodi…  2.45<span style="color: #949494;">e</span>8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> f2fe82f0-4454-4d84-b416-a885f3121e59 e9ac4222-8586-5d9b-8c6d… Decodi…  5.39<span style="color: #949494;">e</span>9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> f2fe82f0-4454-4d84-b416-a885f3121e59 c15acbfd-268a-571c-ab78… Decodi…  1.09<span style="color: #949494;">e</span>9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> f2fe82f0-4454-4d84-b416-a885f3121e59 84e656b7-3740-5fa9-8f81… Decodi…  1.82<span style="color: #949494;">e</span>9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 4d6f6c96-2a83-43d8-8fe1-0f53bffd4674 b1f60da2-db89-556b-b89f… Dissec…  1.18<span style="color: #949494;">e</span>9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> c41dffbf-ad83-447c-a0e1-13e689d9b258 11f8ba42-a556-5757-a0f4… Resolv…  1.30<span style="color: #949494;">e</span>8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> c41dffbf-ad83-447c-a0e1-13e689d9b258 ec8fcbf6-8556-5820-af81… Resolv…  2.90<span style="color: #949494;">e</span>8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> c41dffbf-ad83-447c-a0e1-13e689d9b258 58ff8ba0-280d-51b3-923b… Resolv…  1.25<span style="color: #949494;">e</span>9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 559bb888-7829-41f2-ace5-2c05c7eb81e9 6e061337-810d-5089-a923… Single…  5.11<span style="color: #949494;">e</span>8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 559bb888-7829-41f2-ace5-2c05c7eb81e9 8fbd3d96-dcdf-5c58-b9b5… Single…  2.96<span style="color: #949494;">e</span>8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> 559bb888-7829-41f2-ace5-2c05c7eb81e9 0264fd26-4f63-5a35-a524… Single…  6.29<span style="color: #949494;">e</span>8</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable names ¹​projectTitle, ²​fileSize</span></span></span></code></pre></div>
<p>Choose one project more-or-less arbitrarily, for demonstration
purposes</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">liver_loom_projects</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">projectId</span> <span class="op">==</span> <span class="st">"4d6f6c96-2a83-43d8-8fe1-0f53bffd4674"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 1</span></span>
<span><span class="co">#&gt; Columns: 20</span></span>
<span><span class="co">#&gt; $ projectId                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "4d6f6c96-2a83-43d8-8fe1-0f53bffd4674"</span></span>
<span><span class="co">#&gt; $ projectTitle                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Dissecting the human liver cellular lands…</span></span>
<span><span class="co">#&gt; $ genusSpecies                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">#&gt; $ sampleEntityType            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "specimens"</span></span>
<span><span class="co">#&gt; $ specimenOrgan               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "liver"</span></span>
<span><span class="co">#&gt; $ specimenOrganPart           <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "caudate lobe"</span></span>
<span><span class="co">#&gt; $ selectedCellType            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;&gt;</span></span>
<span><span class="co">#&gt; $ libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "10x 3' v2"</span></span>
<span><span class="co">#&gt; $ nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "single cell"</span></span>
<span><span class="co">#&gt; $ pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> FALSE</span></span>
<span><span class="co">#&gt; $ workflow                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"optimus_post_processing_v1.0.0", "optimu…</span></span>
<span><span class="co">#&gt; $ specimenDisease             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "normal"</span></span>
<span><span class="co">#&gt; $ donorDisease                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "normal"</span></span>
<span><span class="co">#&gt; $ developmentStage            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "human adult stage"</span></span>
<span><span class="co">#&gt; $ fileId                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "b1f60da2-db89-556b-b89f-6fbe29a81697"</span></span>
<span><span class="co">#&gt; $ name                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "sc-landscape-human-liver-10XV2.loom"</span></span>
<span><span class="co">#&gt; $ fileFormat                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "loom"</span></span>
<span><span class="co">#&gt; $ size                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 1176122907</span></span>
<span><span class="co">#&gt; $ version                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2021-02-11T19:41:14.000000Z"</span></span>
<span><span class="co">#&gt; $ url                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "https://service.azul.data.humancellatlas…</span></span></code></pre></div>
<p>Create a filter and download the file</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">liver_loom_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/filters.html" class="external-link">filters</a></span><span class="op">(</span></span>
<span>    projectId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"4d6f6c96-2a83-43d8-8fe1-0f53bffd4674"</span><span class="op">)</span>,</span>
<span>    fileSource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"DCP/2 Analysis"</span><span class="op">)</span>,</span>
<span>    fileFormat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"loom"</span><span class="op">)</span>,</span>
<span>    workflow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="st">"optimus_post_processing_v1.0.0"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">loom_file_path</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files</a></span><span class="op">(</span><span class="va">liver_loom_filter</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="co"># find files matching filter</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files_download</a></span><span class="op">(</span><span class="op">)</span>            <span class="co"># download (and cache) to local disk</span></span></code></pre></div>
<p>The file is cached locally, so running <code><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files_download()</a></code>
a second time does not re-download it.</p>
</div>
<div class="section level3">
<h3 id="integration-with-r-bioconductor-work-flows">Integration with <em>R</em> / <em>Bioconductor</em> work flows<a class="anchor" aria-label="anchor" href="#integration-with-r-bioconductor-work-flows"></a>
</h3>
<p>‘.loom’ files are easily integrated into <em>R</em> /
<em>Bioconductor</em> workflows using the <a href="https://bioconductor.org/packages/LoomExperiment" class="external-link">LoomExperiment</a>
package.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LoomExperiment</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">loom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/export.html" class="external-link">import</a></span><span class="op">(</span><span class="va">loom_file_path</span><span class="op">)</span></span>
<span><span class="va">loom</span></span>
<span><span class="co">#&gt; class: LoomExperiment </span></span>
<span><span class="co">#&gt; dim: 58347 332497 </span></span>
<span><span class="co">#&gt; metadata(15): last_modified CreationDate ...</span></span>
<span><span class="co">#&gt;   project.provenance.document_id specimen_from_organism.organ</span></span>
<span><span class="co">#&gt; assays(1): matrix</span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; rowData names(29): Gene antisense_reads ... reads_per_molecule</span></span>
<span><span class="co">#&gt;   spliced_reads</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; colData names(43): CellID antisense_reads ... reads_unmapped</span></span>
<span><span class="co">#&gt;   spliced_reads</span></span>
<span><span class="co">#&gt; rowGraphs(0): NULL</span></span>
<span><span class="co">#&gt; colGraphs(0): NULL</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata()</a></code> of the loom file contains useful
information on provenance.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">loom</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 15</span></span>
<span><span class="co">#&gt;  $ last_modified                                             : chr "20210211T194047.237912Z"</span></span>
<span><span class="co">#&gt;  $ CreationDate                                              : chr "20210211T193103.553468Z"</span></span>
<span><span class="co">#&gt;  $ LOOM_SPEC_VERSION                                         : chr "3.0.0"</span></span>
<span><span class="co">#&gt;  $ donor_organism.genus_species                              : chr "Homo sapiens"</span></span>
<span><span class="co">#&gt;  $ expression_data_type                                      : chr "exonic"</span></span>
<span><span class="co">#&gt;  $ input_id                                                  : chr "02f207ae-217b-42ce-9733-c03e61541fcc, 2b965070-e2c5-4c26-92c9-c94483f1a00c, 3f9058b9-4243-4ef1-a345-857a5b9ff78"| __truncated__</span></span>
<span><span class="co">#&gt;  $ input_id_metadata_field                                   : chr "sequencing_process.provenance.document_id"</span></span>
<span><span class="co">#&gt;  $ input_name                                                : chr "P5TLH_TLH, P3TLH_TLH, P1TLH_TLH, P4TLH_TLH, P2TLH_TLH"</span></span>
<span><span class="co">#&gt;  $ input_name_metadata_field                                 : chr "sequencing_input.biomaterial_core.biomaterial_id"</span></span>
<span><span class="co">#&gt;  $ library_preparation_protocol.library_construction_approach: chr "10X v2 sequencing"</span></span>
<span><span class="co">#&gt;  $ optimus_output_schema_version                             : chr "1.0.0"</span></span>
<span><span class="co">#&gt;  $ pipeline_version                                          : chr "Optimus_v4.2.2"</span></span>
<span><span class="co">#&gt;  $ project.project_core.project_name                         : chr "SingleCellLiverLandscape"</span></span>
<span><span class="co">#&gt;  $ project.provenance.document_id                            : chr "4d6f6c96-2a83-43d8-8fe1-0f53bffd4674"</span></span>
<span><span class="co">#&gt;  $ specimen_from_organism.organ                              : chr "liver"</span></span></code></pre></div>
<p>The .loom files produced by the standard HCA pipeline contain QC
metrics on the rows (genes) and columns (cells), but no biological
information about the cells. Remedy this using
<code><a href="https://rdrr.io/pkg/hca/man/optimus_loom_annotation.html" class="external-link">optimus_loom_annotation()</a></code></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotated_loom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/optimus_loom_annotation.html" class="external-link">optimus_loom_annotation</a></span><span class="op">(</span><span class="va">loom</span><span class="op">)</span></span>
<span><span class="va">annotated_loom</span></span>
<span><span class="co">#&gt; class: LoomExperiment </span></span>
<span><span class="co">#&gt; dim: 58347 332497 </span></span>
<span><span class="co">#&gt; metadata(16): last_modified CreationDate ...</span></span>
<span><span class="co">#&gt;   specimen_from_organism.organ manifest</span></span>
<span><span class="co">#&gt; assays(1): matrix</span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; rowData names(29): Gene antisense_reads ... reads_per_molecule</span></span>
<span><span class="co">#&gt;   spliced_reads</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; colData names(98): input_id CellID ...</span></span>
<span><span class="co">#&gt;   sequencing_input.biomaterial_core.biomaterial_id</span></span>
<span><span class="co">#&gt;   sequencing_input_type</span></span>
<span><span class="co">#&gt; rowGraphs(0): NULL</span></span>
<span><span class="co">#&gt; colGraphs(0): NULL</span></span>
<span></span>
<span><span class="va">new_columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">annotated_loom</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">loom</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">new_columns</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 55</span></span></code></pre></div>
<p>The following uses names of the annotation columns to show that there
are 5 donors (4 males and one female) ranging in age from 21 to 65 years
old. There are between 39782 and 87538 cells per donor.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">annotated_loom</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span></span>
<span>        biomaterial_id <span class="op">=</span> <span class="va">donor_organism.biomaterial_core.biomaterial_id</span>,</span>
<span>        sex <span class="op">=</span> <span class="va">donor_organism.sex</span>,</span>
<span>        organism_age <span class="op">=</span> <span class="va">donor_organism.organism_age</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 4</span></span></span>
<span><span class="co">#&gt;   biomaterial_id sex    organism_age     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P1TLH          male   44 year      <span style="text-decoration: underline;">39</span>782</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P2TLH          male   65 year      <span style="text-decoration: underline;">73</span>823</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P3TLH          female 41 year      <span style="text-decoration: underline;">66</span>422</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P4TLH          male   21 year      <span style="text-decoration: underline;">64</span>932</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> P5TLH          male   26 year      <span style="text-decoration: underline;">87</span>538</span></span></code></pre></div>
<p>A minimal coercion creates a <a href="https://bioconductor.org/packages/SingleCellExperiment" class="external-link">SingleCellExperiment</a>
that can be used in standard <em>R</em> / <em>Bioconductor</em> single
cell workflows.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">annotated_loom</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 58347 332497 </span></span>
<span><span class="co">#&gt; metadata(16): last_modified CreationDate ...</span></span>
<span><span class="co">#&gt;   specimen_from_organism.organ manifest</span></span>
<span><span class="co">#&gt; assays(1): matrix</span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; rowData names(29): Gene antisense_reads ... reads_per_molecule</span></span>
<span><span class="co">#&gt;   spliced_reads</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; colData names(98): input_id CellID ...</span></span>
<span><span class="co">#&gt;   sequencing_input.biomaterial_core.biomaterial_id</span></span>
<span><span class="co">#&gt;   sequencing_input_type</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] LoomExperiment_1.15.0       BiocIO_1.7.1               </span></span>
<span><span class="co">#&gt;  [3] rhdf5_2.41.1                SingleCellExperiment_1.19.0</span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.27.1 Biobase_2.57.1             </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.49.0        GenomeInfoDb_1.33.3        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.31.0              MatrixGenerics_1.9.1       </span></span>
<span><span class="co">#&gt; [11] matrixStats_0.62.0          S4Vectors_0.35.1           </span></span>
<span><span class="co">#&gt; [13] BiocGenerics_0.43.0         hca_1.5.5                  </span></span>
<span><span class="co">#&gt; [15] dplyr_1.0.9                </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] bitops_1.0-7           fs_1.5.2               bit64_4.0.5           </span></span>
<span><span class="co">#&gt;  [4] filelock_1.0.2         httr_1.4.3             rprojroot_2.0.3       </span></span>
<span><span class="co">#&gt;  [7] tools_4.2.0            bslib_0.4.0            utf8_1.2.2            </span></span>
<span><span class="co">#&gt; [10] R6_2.5.1               DT_0.23                HDF5Array_1.25.1      </span></span>
<span><span class="co">#&gt; [13] DBI_1.1.3              rhdf5filters_1.9.0     tidyselect_1.1.2      </span></span>
<span><span class="co">#&gt; [16] bit_4.0.4              curl_4.3.2             compiler_4.2.0        </span></span>
<span><span class="co">#&gt; [19] textshaping_0.3.6      cli_3.3.0              formatR_1.12          </span></span>
<span><span class="co">#&gt; [22] desc_1.4.1             DelayedArray_0.23.0    sass_0.4.2            </span></span>
<span><span class="co">#&gt; [25] readr_2.1.2            rappdirs_0.3.3         pkgdown_2.0.6         </span></span>
<span><span class="co">#&gt; [28] systemfonts_1.0.4      stringr_1.4.0          digest_0.6.29         </span></span>
<span><span class="co">#&gt; [31] rmarkdown_2.14         XVector_0.37.0         pkgconfig_2.0.3       </span></span>
<span><span class="co">#&gt; [34] htmltools_0.5.3        dbplyr_2.2.1           fastmap_1.1.0         </span></span>
<span><span class="co">#&gt; [37] htmlwidgets_1.5.4      rlang_1.0.4            RSQLite_2.2.15        </span></span>
<span><span class="co">#&gt; [40] shiny_1.7.2            jquerylib_0.1.4        generics_0.1.3        </span></span>
<span><span class="co">#&gt; [43] jsonlite_1.8.0         vroom_1.5.7            RCurl_1.98-1.7        </span></span>
<span><span class="co">#&gt; [46] magrittr_2.0.3         GenomeInfoDbData_1.2.8 futile.logger_1.4.3   </span></span>
<span><span class="co">#&gt; [49] Matrix_1.4-1           Rcpp_1.0.9             Rhdf5lib_1.19.2       </span></span>
<span><span class="co">#&gt; [52] fansi_1.0.3            lifecycle_1.0.1        stringi_1.7.8         </span></span>
<span><span class="co">#&gt; [55] yaml_2.3.5             zlibbioc_1.43.0        BiocFileCache_2.5.0   </span></span>
<span><span class="co">#&gt; [58] grid_4.2.0             blob_1.2.3             parallel_4.2.0        </span></span>
<span><span class="co">#&gt; [61] promises_1.2.0.1       crayon_1.5.1           miniUI_0.1.1.1        </span></span>
<span><span class="co">#&gt; [64] lattice_0.20-45        hms_1.1.1              knitr_1.39            </span></span>
<span><span class="co">#&gt; [67] pillar_1.8.0           futile.options_1.0.1   glue_1.6.2            </span></span>
<span><span class="co">#&gt; [70] evaluate_0.15          lambda.r_1.2.4         vctrs_0.4.1           </span></span>
<span><span class="co">#&gt; [73] tzdb_0.3.0             httpuv_1.6.5           purrr_0.3.4           </span></span>
<span><span class="co">#&gt; [76] tidyr_1.2.0            assertthat_0.2.1       cachem_1.0.6          </span></span>
<span><span class="co">#&gt; [79] xfun_0.31              mime_0.12              xtable_1.8-4          </span></span>
<span><span class="co">#&gt; [82] later_1.3.0            ragg_1.2.2             tibble_3.1.7          </span></span>
<span><span class="co">#&gt; [85] memoise_2.0.1          ellipsis_0.3.2</span></span></code></pre></div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>Roswell Park Comprehensive Cancer Center (RPCCC)<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Roswell Park Comprehensive Cancer Center (RPCCC)<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Martin Morgan, Maya McDaniel, Kayla Interdonato, Yubo Cheng.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
