<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>2. Accessing CellxGene Data From Bioconductor • ExploringHCACxG</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="2. Accessing CellxGene Data From Bioconductor">
<meta property="og:description" content="ExploringHCACxG">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ExploringHCACxG</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A_HCA.html">1. Accessing Human Cell Atlas Data From Bioconductor</a>
    </li>
    <li>
      <a href="../articles/B_CXG.html">2. Accessing CellxGene Data From Bioconductor</a>
    </li>
    <li>
      <a href="../articles/C_AnVIL.html">3. Accessing Human Cell Atlas Data on the AnVIL Cloud</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mtmorgan/ExploringHCACxG">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>2. Accessing CellxGene Data From
Bioconductor</h1>
                        <h4 data-toc-skip class="author">Martin Morgan<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, Kayla
Interdonato<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, Yubo Cheng<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/ExploringHCACxG/blob/HEAD/vignettes/B_CXG.Rmd"><code>vignettes/B_CXG.Rmd</code></a></small>
      <div class="hidden name"><code>B_CXG.Rmd</code></div>

    </div>

    
    
<p>Last modified: 20 July, 2022</p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>FIXME: Along with the topic of your workshop, include how students
can expect to spend their time. For the description may also include
information about what type of workshop it is (e.g. instructor-led live
demo, lab, lecture + lab, etc.). Instructors are strongly recommended to
provide completely worked examples for lab sessions, and a set of
stand-alone notes that can be read and understood outside of the
workshop.</p>
</div>
<div class="section level3">
<h3 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h3>
<p>The focus is on <a href="https://bioconductor.org/packages/cellxgenedp" class="external-link">cellxgenedp</a>
package. This package emphasize <a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a> and ‘tidy’
approaches to working with data.frames. Files downloaded from the HCA or
CellXGene web sites can be imported into R / Bioconductor as <a href="https://bioconductor.org/packages/SingleCellExperiment" class="external-link">SingleCellExperiment</a>
objects through the <a href="https://bioconductor.org/packages/LoomExperiment" class="external-link">LoomExperiment</a>
and <a href="https://bioconductor.org/packages/zellkonverter" class="external-link">zellkonverter</a>
packages.</p>
</div>
<div class="section level3">
<h3 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction to the CellXGene data browser</td>
<td>4m</td>
</tr>
<tr class="even">
<td>CellXGene data discovery in R / Bioconductor</td>
<td>7m</td>
</tr>
<tr class="odd">
<td>Integration with R / Bioconductor single-cell analysis</td>
<td>4m</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="workshop-goals-and-objectives">Workshop goals and objectives<a class="anchor" aria-label="anchor" href="#workshop-goals-and-objectives"></a>
</h3>
<p>List “big picture” student-centered workshop goals and learning
objectives. Learning goals and objectives are related, but not the same
thing. These goals and objectives will help some people to decide
whether to attend the conference for training purposes, so please make
these as precise and accurate as possible.</p>
<p><em>Learning goals</em> are high-level descriptions of what
participants will learn and be able to do after the workshop is over.
<em>Learning objectives</em>, on the other hand, describe in very
specific and measurable terms specific skills or knowledge attained. The
<a href="#bloom">Bloom’s Taxonomy</a> may be a useful framework for
defining and describing your goals and objectives, although there are
others.</p>
</div>
<div class="section level3">
<h3 id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h3>
<p>Some examples:</p>
<ul>
<li>describe how to…</li>
<li>identify methods for…</li>
<li>understand the difference between…</li>
</ul>
</div>
<div class="section level3">
<h3 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h3>
<ul>
<li>analyze xyz data to produce…</li>
<li>create xyz plots</li>
<li>evaluate xyz data for artifacts</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="workshop">Workshop<a class="anchor" aria-label="anchor" href="#workshop"></a>
</h2>
<div class="section level3">
<h3 id="exploring-the-cellxgene-web-sites">Exploring the CellXGene web sites<a class="anchor" aria-label="anchor" href="#exploring-the-cellxgene-web-sites"></a>
</h3>
<p>Datasets</p>
<ul>
<li><a href="https://cellxgene.cziscience.com/" class="external-link uri">https://cellxgene.cziscience.com/</a></li>
</ul>
<p>Interactive visualization</p>
<ul>
<li>Easy visualization is pretty cool!</li>
</ul>
<p>Some observations</p>
<ul>
<li>Why is visualization fast? Pre-computed dimensionality
reduction.</li>
<li>What workflow was used in processing data? It’s up to the
contributor, and not documented.</li>
</ul>
</div>
<div class="section level3">
<h3 id="working-with-cellxgene-data">Working with CellXGene data<a class="anchor" aria-label="anchor" href="#working-with-cellxgene-data"></a>
</h3>
<p>Load the packages we’ll use</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cellxgenedp</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/theislab/zellkonverter" class="external-link">zellkonverter</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Connect to the database</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/db.html" class="external-link">db</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Discover datasets with 10x assays of African American females</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">african_american_female</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/query.html" class="external-link">datasets</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/facets.html" class="external-link">facets_filter</a></span><span class="op">(</span><span class="va">assay</span>, <span class="st">"ontology_term_id"</span>, <span class="st">"EFO:0009922"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/facets.html" class="external-link">facets_filter</a></span><span class="op">(</span><span class="va">ethnicity</span>, <span class="st">"label"</span>, <span class="st">"African American"</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/facets.html" class="external-link">facets_filter</a></span><span class="op">(</span><span class="va">sex</span>, <span class="st">"label"</span>, <span class="st">"female"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">african_american_female</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 21 × 27</span></span></span>
<span><span class="co">#&gt;    dataset_id     colle…¹ assay  cell_…² cell_…³ datas…⁴ devel…⁵ disease ethni…⁶</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> de985818-285f… c9706a… &lt;list&gt;   <span style="text-decoration: underline;">31</span>696 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> f72958f5-7f42… 2f75d2… &lt;list&gt;  <span style="text-decoration: underline;">982</span>538 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 07854d9c-5375… bcb614… &lt;list&gt;  <span style="text-decoration: underline;">172</span>847 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 0b75c598-0893… bcb614… &lt;list&gt;  <span style="text-decoration: underline;">304</span>652 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 32b9bdce-2481… bcb614… &lt;list&gt;  <span style="text-decoration: underline;">107</span>344 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 59b69042-47c2… b9fc3d… &lt;list&gt;   <span style="text-decoration: underline;">49</span>139 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> e763ed0d-0e5a… b9fc3d… &lt;list&gt;    <span style="text-decoration: underline;">7</span>274 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> db0752b9-f20e… b9fc3d… &lt;list&gt;   <span style="text-decoration: underline;">55</span>348 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> d9b4bc69-ed90… b9fc3d… &lt;list&gt;   <span style="text-decoration: underline;">20</span>000 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> bc2a7b3d-f04e… b9fc3d… &lt;list&gt;  <span style="text-decoration: underline;">109</span>995 &lt;list&gt;  https:… &lt;list&gt;  &lt;list&gt;  &lt;list&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 11 more rows, 18 more variables: is_primary_data &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   is_valid &lt;lgl&gt;, linked_genesets &lt;lgl&gt;, mean_genes_per_cell &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   name &lt;chr&gt;, organism &lt;list&gt;, processing_status &lt;list&gt;, published &lt;lgl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   revision &lt;int&gt;, schema_version &lt;chr&gt;, sex &lt;list&gt;, tissue &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   tombstone &lt;lgl&gt;, x_normalization &lt;chr&gt;, created_at &lt;date&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   published_at &lt;date&gt;, revised_at &lt;date&gt;, updated_at &lt;date&gt;, and abbreviated</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variable names ¹​collection_id, ²​cell_count, ³​cell_type, …</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>Find H5AD files for these datasets; visualize and download one for
illustration</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_files</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">african_american_female</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">dataset_id</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/query.html" class="external-link">files</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span>,</span>
<span>        by <span class="op">=</span> <span class="st">"dataset_id"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">selected_files</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 63 × 8</span></span></span>
<span><span class="co">#&gt;    dataset_id       file_id filen…¹ filet…² s3_uri user_…³ created_at updated_at</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> de985818-285f-4… d5c6de… local.… H5AD    s3://… TRUE    2021-09-24 2021-09-24</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> de985818-285f-4… a0bcae… local.… RDS     s3://… TRUE    2021-09-24 2021-12-21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> de985818-285f-4… dff842… explor… CXG     s3://… TRUE    2021-09-24 2021-09-24</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> f72958f5-7f42-4… 2e7374… local.… RDS     s3://… TRUE    2021-10-07 2021-12-21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> f72958f5-7f42-4… bbce34… explor… CXG     s3://… TRUE    2021-10-07 2021-10-07</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> f72958f5-7f42-4… 091323… local.… H5AD    s3://… TRUE    2021-10-07 2021-10-07</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 07854d9c-5375-4… f6f812… local.… H5AD    s3://… TRUE    2022-02-15 2022-02-15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 07854d9c-5375-4… 7ae7df… explor… CXG     s3://… TRUE    2022-02-15 2022-02-15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 07854d9c-5375-4… 3f995f… local.… RDS     s3://… TRUE    2022-02-15 2022-02-15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 0b75c598-0893-4… 38eaca… local.… H5AD    s3://… TRUE    2022-02-15 2022-02-15</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 53 more rows, and abbreviated variable names ¹​filename, ²​filetype,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ³​user_submitted</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
<p>Visualize ‘CXG’ files in the CellxGene web browser</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">filetype</span> <span class="op">==</span> <span class="st">"CXG"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/query.html" class="external-link">datasets_visualize</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Download ‘H5AD’ (anndata) files for local use</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5ad_file</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">selected_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">filetype</span> <span class="op">==</span> <span class="st">"H5AD"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/cellxgenedp/man/query.html" class="external-link">files_download</a></span><span class="op">(</span>dry.run <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="importing--h5ad-data-into-r-bioconductor">Importing .h5ad data into <em>R</em> / <em>Bioconductor</em><a class="anchor" aria-label="anchor" href="#importing--h5ad-data-into-r-bioconductor"></a>
</h3>
<p>Read the H5AD file into <em>R</em></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h5ad</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/zellkonverter/man/readH5AD.html" class="external-link">readH5AD</a></span><span class="op">(</span><span class="va">h5ad_file</span>, reader <span class="op">=</span> <span class="st">"R"</span>, use_hdf5 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">h5ad</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 33234 31696 </span></span>
<span><span class="co">#&gt; metadata(5): X_normalization default_embedding layer_descriptions</span></span>
<span><span class="co">#&gt;   schema_version title</span></span>
<span><span class="co">#&gt; assays(1): X</span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; rowData names(4): feature_biotype feature_is_filtered feature_name</span></span>
<span><span class="co">#&gt;   feature_reference</span></span>
<span><span class="co">#&gt; colnames(31696): CMGpool_AAACCCAAGGACAACC CMGpool_AAACCCACAATCTCTT ...</span></span>
<span><span class="co">#&gt;   K109064_TTTGTTGGTTGCATCA K109064_TTTGTTGGTTGGACCC</span></span>
<span><span class="co">#&gt; colData names(34): assay assay_ontology_term_id ...</span></span>
<span><span class="co">#&gt;   tissue_ontology_term_id tyrer_cuzick_lifetime_risk</span></span>
<span><span class="co">#&gt; reducedDimNames(2): X_tsne X_umap</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="integration-with-r-bioconductor-work-flows">Integration with <em>R</em> / <em>Bioconductor</em> work flows<a class="anchor" aria-label="anchor" href="#integration-with-r-bioconductor-work-flows"></a>
</h3>
<p>Work with the data, e.g., simple visualization to illustrate the
familiar <em>R</em> / <em>Bioconductor</em> tools ‘just work’.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">umap</span> <span class="op">=</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">h5ad</span>, <span class="st">"X_umap"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The `x` argument of `as_tibble.matrix()` must have unique column names if `.name_repair` is omitted as of tibble 2.0.0.</span></span>
<span><span class="co">#&gt; Using compatibility `.name_repair`.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</span></span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">umap</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">V1</span>, y <span class="op">=</span> <span class="va">V2</span>, color <span class="op">=</span> <span class="va">h5ad</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>pch <span class="op">=</span> <span class="st">"."</span><span class="op">)</span></span></code></pre></div>
<p><img src="B_CXG_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggplot2_3.3.6               SingleCellExperiment_1.19.0</span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.27.1 Biobase_2.57.1             </span></span>
<span><span class="co">#&gt;  [5] GenomicRanges_1.49.0        GenomeInfoDb_1.33.3        </span></span>
<span><span class="co">#&gt;  [7] IRanges_2.31.0              S4Vectors_0.35.1           </span></span>
<span><span class="co">#&gt;  [9] BiocGenerics_0.43.0         MatrixGenerics_1.9.1       </span></span>
<span><span class="co">#&gt; [11] matrixStats_0.62.0          zellkonverter_1.7.3        </span></span>
<span><span class="co">#&gt; [13] dplyr_1.0.9                 cellxgenedp_1.1.2          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] bitops_1.0-7           fs_1.5.2               filelock_1.0.2        </span></span>
<span><span class="co">#&gt;  [4] httr_1.4.3             rprojroot_2.0.3        tools_4.2.0           </span></span>
<span><span class="co">#&gt;  [7] bslib_0.4.0            utf8_1.2.2             R6_2.5.1              </span></span>
<span><span class="co">#&gt; [10] DT_0.23                HDF5Array_1.25.1       DBI_1.1.3             </span></span>
<span><span class="co">#&gt; [13] colorspace_2.0-3       rhdf5filters_1.9.0     withr_2.5.0           </span></span>
<span><span class="co">#&gt; [16] tidyselect_1.1.2       curl_4.3.2             compiler_4.2.0        </span></span>
<span><span class="co">#&gt; [19] textshaping_0.3.6      cli_3.3.0              basilisk.utils_1.9.1  </span></span>
<span><span class="co">#&gt; [22] desc_1.4.1             DelayedArray_0.23.0    labeling_0.4.2        </span></span>
<span><span class="co">#&gt; [25] sass_0.4.2             scales_1.2.0           pkgdown_2.0.6         </span></span>
<span><span class="co">#&gt; [28] systemfonts_1.0.4      stringr_1.4.0          digest_0.6.29         </span></span>
<span><span class="co">#&gt; [31] rmarkdown_2.14         basilisk_1.9.2         XVector_0.37.0        </span></span>
<span><span class="co">#&gt; [34] pkgconfig_2.0.3        htmltools_0.5.3        fastmap_1.1.0         </span></span>
<span><span class="co">#&gt; [37] highr_0.9              htmlwidgets_1.5.4      rlang_1.0.4           </span></span>
<span><span class="co">#&gt; [40] shiny_1.7.2            jquerylib_0.1.4        generics_0.1.3        </span></span>
<span><span class="co">#&gt; [43] farver_2.1.0           jsonlite_1.8.0         RCurl_1.98-1.7        </span></span>
<span><span class="co">#&gt; [46] magrittr_2.0.3         GenomeInfoDbData_1.2.8 Matrix_1.4-1          </span></span>
<span><span class="co">#&gt; [49] Rcpp_1.0.9             munsell_0.5.0          Rhdf5lib_1.19.2       </span></span>
<span><span class="co">#&gt; [52] fansi_1.0.3            reticulate_1.25        lifecycle_1.0.1       </span></span>
<span><span class="co">#&gt; [55] stringi_1.7.8          yaml_2.3.5             zlibbioc_1.43.0       </span></span>
<span><span class="co">#&gt; [58] rhdf5_2.41.1           grid_4.2.0             parallel_4.2.0        </span></span>
<span><span class="co">#&gt; [61] promises_1.2.0.1       crayon_1.5.1           dir.expiry_1.5.0      </span></span>
<span><span class="co">#&gt; [64] lattice_0.20-45        knitr_1.39             pillar_1.8.0          </span></span>
<span><span class="co">#&gt; [67] glue_1.6.2             evaluate_0.15          vctrs_0.4.1           </span></span>
<span><span class="co">#&gt; [70] png_0.1-7              httpuv_1.6.5           gtable_0.3.0          </span></span>
<span><span class="co">#&gt; [73] purrr_0.3.4            assertthat_0.2.1       cachem_1.0.6          </span></span>
<span><span class="co">#&gt; [76] xfun_0.31              mime_0.12              xtable_1.8-4          </span></span>
<span><span class="co">#&gt; [79] later_1.3.0            ragg_1.2.2             tibble_3.1.7          </span></span>
<span><span class="co">#&gt; [82] memoise_2.0.1          ellipsis_0.3.2</span></span></code></pre></div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>Roswell Park Comprehensive Cancer Center (RPCCC)<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>RPCCC<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>RPCCC<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Martin Morgan, Maya McDaniel, Kayla Interdonato, Yubo Cheng.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
